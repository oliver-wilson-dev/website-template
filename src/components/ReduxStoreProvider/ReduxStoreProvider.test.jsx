import React from 'react';
import { shallow } from 'enzyme';
import thunk from 'redux-thunk';

import reducer from '../../state/reducers';

const render = () => {
  const ReduxStoreProvider = require('./ReduxStoreProvider').default;

  return shallow(<ReduxStoreProvider />);
};

let createStore;
let applyMiddleware;
let composeWithDevTools;
let Provider;

describe('ReduxStoreProvider', () => {
  const mockStore = Symbol('test-store');
  const mockComposeWithDevTools = Symbol('test-devtools');
  const mockApplyMiddleware = Symbol('test-apply-middleware');
  const mockReducer = Symbol('test-reducer');

  beforeEach(() => {
    jest.resetModules();
    jest.doMock('react-redux', () => ({
      Provider: ({ children }) => <>{children}</>
    }));

    jest.doMock('redux', () => ({
      createStore: jest.fn(),
      applyMiddleware: jest.fn()
    }));

    jest.doMock('../../state/reducers', () => ({
      __esModule: true,
      default: mockReducer
    }));

    jest.doMock('redux-devtools-extension', () => ({
      composeWithDevTools: jest.fn()
    }));

    jest.doMock('redux-thunk', () => ({
      __esModule: true,
      default: 'test-thunk'
    }));

    ({ createStore, applyMiddleware } = require('redux'));
    ({ composeWithDevTools } = require('redux-devtools-extension'));
    ({ Provider } = require('react-redux'));

    createStore.mockReturnValue(mockStore);
    composeWithDevTools.mockReturnValue(mockComposeWithDevTools);
    applyMiddleware.mockReturnValue(mockApplyMiddleware);
  });

  it('should call createStore with the reducer and the composed middleware', () => {
    render();

    expect(createStore).toHaveBeenCalledWith(mockReducer, mockComposeWithDevTools);
  });

  it('should call composeWithDevTools with the result of applyMiddleware', () => {
    render();

    expect(composeWithDevTools).toHaveBeenCalledWith(mockApplyMiddleware);
  });

  it('should render correctly', () => {
    expect(render()).toMatchSnapshot();
  });

  it('should pass the store to the provider', () => {
    expect(render().find(Provider).prop('store')).toBe(mockStore);
  });
});
