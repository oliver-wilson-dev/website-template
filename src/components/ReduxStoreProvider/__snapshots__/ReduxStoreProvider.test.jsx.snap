// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReduxStoreProvider should render correctly 1`] = `
<Provider
  store={Symbol(test-store)}
/>
`;
